cogs/get_token_authorisation.py

```diff
@@ -0,0 +1,108 @@
+"""Contains cog classes for token authorisation check interactions."""
+
+from collections.abc import Sequence
+from typing import Final
+
+__all__: Sequence[str] = ("GetTokenAuthorisationSlashCommand",)
+
+import logging
+from logging import Logger
+
+import aiohttp
+import bs4
+import discord
+from bs4 import BeautifulSoup
+
+from config import settings
+from utils import CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class GetTokenAuthorisationSlashCommand(TeXBotBaseCog):
+    """Cog class that defines the /get_token_authorisation command."""
+
+    @discord.slash_command( # type: ignore[no-untyped-call, misc]
+        name="get_token_authorisation",
+        description="Checks the authorisations held by the token.",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def get_token_authorisation(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition of the get token authorisation command.
+
+        The command will retrieve the profle for the user who owns the token.
+        The profile page will contain the users name and a list of the MSL organisations
+        the user has administrative access to.
+        """
+        request_headers: dict[str, str] = {
+            "Cache-Control": "no-cache",
+            "Pragma": "no-cache",
+            "Expires": "0",
+        }
+
+        request_cookies: dict[str, str] = {
+            ".ASPXAUTH": settings["MEMBERS_LIST_URL_SESSION_COOKIE"],
+        }
+
+        REQUEST_URL: Final[str] = "https://guildofstudents.com/profile"
+
+        async with aiohttp.ClientSession(headers=request_headers, cookies=request_cookies) as http_session:  # noqa: E501, SIM117
+            async with http_session.get(REQUEST_URL) as http_response:
+                response_html: str = await http_response.text()
+
+        TABLE_ID: str = "ulOrgs"
+        parsed_html: bs4.Tag | bs4.NavigableString | None = BeautifulSoup(
+            response_html,
+            "html.parser",
+        ).find(
+            "ul",
+            {"id": TABLE_ID},
+        )
```

Comment:
Can be simplified onto a single line: `).find("ul", {"id": TABLE_ID})` or `).find("ul", {"id": "ulOrgs"})` according to above decision